### ISSUE TITLE
Parsing error on "ALTER TABLE ... RENAME COLUMN ..."

### ISSUE THREAD

**Issue by davidfuhr** [sqlfluff/sqlfluff#3594]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I got a parsing error with the following sql:

```sql
ALTER TABLE employee RENAME COLUMN team_number TO team_id;
```

### Expected Behaviour

No parsing error.

### Observed Behaviour

Parsing error:

```
L:   1 | P:  36 |  PRS | Line 1, Position 36: Found unparsable section:
                       | 'team_number TO team_id'
```

### How to reproduce

```sql
ALTER TABLE employee RENAME COLUMN team_number TO team_id;
```

```
❯ sqlfluff lint alter-table-rename-column.sql
== [alter-table-rename-column.sql] FAIL                                                                                                                                                                     
L:   1 | P:  36 |  PRS | Line 1, Position 36: Found unparsable section:
                       | 'team_number TO team_id'
WARNING: Parsing errors found and dialect is set to 'oracle'. Have you configured your dialect correctly?
All Finished 📜 🎉!

```

### Dialect

oracle

### Version

```
❯ python3 --version
Python 3.9.13ersion
❯ sqlfluff --version
sqlfluff, version 1.1.0
```

### Configuration

[sqlfluff]
dialect = oracle
rules = L029,L038,L049,L052,L057

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
